src/modalities/nn/weight_init/low_level_weight_init_factory.py

```diff
@@ -0,0 +1,121 @@
+import math
+from typing import Annotated, List, Optional
+
+from pydantic import BaseModel, Field, model_validator
+
+from modalities.config.pydanctic_if_types import PydanticWeightInitializationIFType
+from modalities.nn.weight_init.weight_init import (
+    ModulewiseNormalInitialization,
+    NamedParameterwiseNormalInitialization,
+    WeightInitializationIF,
+    WeightInitializerWrapper,
+)
+
+
+class WeightInitializerWrapperConfig(BaseModel):
+    weight_initializers: List[PydanticWeightInitializationIFType]
+
+
+class PlainWeightInitializationConfig(BaseModel):
+    mean: float
+    std: Annotated[float, Field(strict=True, ge=0.0)] | str  # can be float or "auto"
+    hidden_dim: Optional[int] = None
+
+    @model_validator(mode="after")
+    def check_std_and_hidden_dim(self):
+        if self.std == "auto" and self.hidden_dim is None:
+            raise ValueError("hidden_dim must be specified when std is 'auto'")
+        return self
+
+
+class NamedParameterwiseNormalInitializationConfig(BaseModel):
+    mean: float
+    std: Annotated[float, Field(strict=True, ge=0.0)] | str  # can be float or "auto"
+    parameter_name_suffixes: List[str]  # here we filter for the parameter names, e.g., "c_proj.weight"
+
+
+class ScaledWeightInitializationConfig(BaseModel):
+    mean: float
+    plain_std: Annotated[float, Field(strict=True, ge=0.0)]
+    num_layers: Annotated[int, Field(strict=True, gt=0)]
+    parameter_name_suffixes: List[str]  # here we filter for the parameter names, e.g., "c_proj.weight"
+
+
+class ScaledEmbedInitializationConfig(BaseModel):
+    mean: float
+    parameter_name_suffixes: List[str]  # here we filter for the parameter names, e.g., "c_proj.weight"
+
+
+class LowLevelInitializationFactory:
+    @staticmethod
+    def get_weight_initializer_wrapper(weight_initializers: List[WeightInitializationIF]) -> WeightInitializationIF:
```

Comment:
This function is used only in `HighLevelWeightInitializationFactory`, and I think it would make more sense to move it there. One might even just get rid of it and use the `WeightInitializerWrapper` class directly, as a simplification.

Reply:
done